pkg/pillar/cmd/zedrouter/appcontainer.go

```diff
@@ -328,3 +336,178 @@ func (z *zedrouter) getAppContainers(status types.AppNetworkStatus) (
+			z.log.Errorf("getNestedDomainAppMetrics: failed to decode JSON data, error: %v", err)
+			continue
+		}
+
+		acStats := types.AppContainerStats{
+			ContainerName:  nastat.Id,
+			Status:         nastat.Status,
+			Pids:           nastat.Pids,
+			Uptime:         nastat.Uptime,
+			CPUTotal:       nastat.CPUTotal,
+			SystemCPUTotal: nastat.SystemCPUTotal,
+			UsedMem:        nastat.UsedMem,
+			AllocatedMem:   nastat.AllocatedMem,
+			TxBytes:        nastat.TxBytes,
+			RxBytes:        nastat.RxBytes,
+			ReadBytes:      nastat.ReadBytes,
+			WriteBytes:     nastat.WriteBytes,
+		}
+		acMetrics.StatsList = append(acMetrics.StatsList, acStats)
+	}
+	// send for zedagent to pack w/ parent app metrics
+	z.pubAppContainerStats.Publish(acMetrics.Key(), acMetrics)
+
+	z.log.Functionf("getNestedDomainAppMetrics: collected metrics %+v", acMetrics)
+}
+
+// getNestedDomainAppList gets the list of nested domain apps
+func (z *zedrouter) getNestedDomainAppList(status types.AppNetworkStatus) ([]types.NestedAppDomainStatus, error) {
+	pub := z.pubNestedAppDomainStatus
+	existingItems := pub.GetAll()
+	existingNestedApps := make(map[string]types.NestedAppDomainStatus)
+
+	// Save existing items for later comparison
+	for _, st := range existingItems {
+		nestedApp := st.(types.NestedAppDomainStatus)
+		existingNestedApps[nestedApp.UUIDandVersion.UUID.String()] = nestedApp
+	}
+
+	// Get the JSON data from the Runtime endpoint
+	url := "http://" + status.GetStatsIPAddr.String() + ":" + strconv.Itoa(nestedAppDomainAppPort) +
+		nestedAppDomainAppListURL
+	data, err := fetchHTTPData(url)
+	if err != nil {
+		z.log.Errorf("getNestedDomainAppMetrics: %v", err)
+		return nil, err
+	}
+
+	var nestedAppInventory nestedapp.NestedAppInventory
+	// Decode the JSON data into the protobuf struct
+	if err := protojson.Unmarshal(data, &nestedAppInventory); err != nil {
+		z.log.Errorf("getNestedAppListAndMetrics: failed to decode JSON data using protojson, error: %v", err)
+		return nil, err
+	}
+
+	// Process the nested app IDs
+	var nestedapps []types.NestedAppDomainStatus
+	newNestedApps := make(map[string]types.NestedAppDomainStatus)
+	for _, nestedAppID := range nestedAppInventory.Apps {
+		nestedAppUUID, err := uuid.FromString(nestedAppID.AppId)
+		if err != nil {
+			z.log.Errorf("getNestedAppListAndMetrics: invalid UUID %s, error: %v", nestedAppID.AppId, err)
+			continue
+		}
+
+		nestedApp := types.NestedAppDomainStatus{
+			UUIDandVersion: types.UUIDandVersion{UUID: nestedAppUUID},
+			DisplayName:    nestedAppID.AppName,
+			DisableLogs:    nestedAppID.DisableLogs,
+			ParentAppUUID:  status.UUIDandVersion.UUID,
+		}
+
+		newNestedApps[nestedAppID.AppId] = nestedApp
+		nestedapps = append(nestedapps, nestedApp)
+	}
+
+	// Compare old and new sets of nested apps and publish if different
+	for uuidStr, newNestedApp := range newNestedApps {
+		if existingNestedApp, exists := existingNestedApps[uuidStr]; !exists || !cmp.Equal(existingNestedApp, newNestedApp) {
```

Comment:
is it really possible that 2 apps share the same UUID? do we need `!cmp.Equal(existingNestedApp, newNestedApp)` check here?

Reply:
i think the only item in the structure is the 'DisableLogs' which might change. it compares two structures, yes, they are not likely to change inside.